From f7efa15340900ddc28ca9baf36354af5798a9f4d Mon Sep 17 00:00:00 2001
From: Josh Durgin <josh.durgin@inktank.com>
Date: Mon, 7 Apr 2014 16:52:03 -0700
Subject: [PATCH 170/213] rbd: set the remaining discard properties to enable
 support
References: fate#318918
Git-commit: b76f82398c1017e303d87760e22125714010207f
Patch-mainline: v3.18-rc1

max_discard_sectors must be set for the queue to support discard.
Operations implementing discard for rbd zero data, so report that.

Signed-off-by: Josh Durgin <josh.durgin@inktank.com>
Acked-by: David Disseldorp <ddiss@suse.de>

---
 drivers/block/rbd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/block/rbd.c b/drivers/block/rbd.c
index 62f5f22..dbe15be 100644
--- a/drivers/block/rbd.c
+++ b/drivers/block/rbd.c
@@ -3806,6 +3806,8 @@ static int rbd_init_disk(struct rbd_device *rbd_dev)
 	queue_flag_set_unlocked(QUEUE_FLAG_DISCARD, q);
 	q->limits.discard_granularity = segment_size;
 	q->limits.discard_alignment = segment_size;
+	q->limits.max_discard_sectors = segment_size / SECTOR_SIZE;
+	q->limits.discard_zeroes_data = 1;
 
 	blk_queue_merge_bvec(q, rbd_merge_bvec);
 	disk->queue = q;
-- 
2.1.4

